backend/corpora/lambdas/api/v1/collection.py

```diff
@@ -71,12 +71,21 @@ def get_collection_dataset(dataset_uuid: str):
 def delete_collection(collection_uuid: str, user: str):
     db_session = g.db_session
     collection = Collection.get_collection(
-        db_session, collection_uuid, CollectionVisibility.PRIVATE.name, include_tombstones=True
+        db_session, collection_uuid, CollectionVisibility.PRIVATE.name, include_tombstones=True, owner=user
     )
     if not collection:
         raise ForbiddenHTTPException()
-    if not Collection.if_owner(db_session, collection.id, collection.visibility, user):
-        raise ForbiddenHTTPException()
     if not collection.tombstone:
         collection.tombstone_collection()
     return "", 202
+
+
+def update_collection(collection_uuid: str, body: dict, user: str):
+    db_session = g.db_session
+    collection = Collection.get_collection(db_session, collection_uuid, CollectionVisibility.PRIVATE.name, owner=user)
```

Comment:
feels like we just need to check if the user has write access to the collection before updating the collection here, instead of restricting the collection access to `WRITE only`?

Reply:
At this point, we are not updating the collection. We are only fetching it from the DB. If the user is the owner of the collection then they have write access.

Reply:
Oh I didn't realize `owner` argument was optional 😆  

My thought was instead of figuring out when to pass `owner` as an optional argument to `Collection.get_collection` as a way to indirectly check if the user is the owner, would it be more straightforward to just do:

```py
collection = Collection.get_collection(db_session, collection_uuid, CollectionVisibility.PRIVATE.name)

if not (check user is the owner of the collection):
  raise ForbiddenHTTPException()

# update collection here
```
or maybe update description of `Collection.get_collection()` to include notes that `owner` argument is optional and is only needed if the callsite needs to return WRITE only collection?

Reply:
By including the owner in the filter we are performing the check for ownership in the query. I'll update the documentation.

Reply:
Awesome thank you!